fastcrypto/src/secp256r1/mod.rs

```diff
@@ -333,17 +330,12 @@ impl Secp256r1KeyPair {
     /// Create a new signature using the given hash function to hash the message.
     pub fn sign_with_hash<H: HashFunction<32>>(&self, msg: &[u8]) -> Secp256r1Signature {
         // Private key as scalar
-        let x = self.secret.privkey.as_nonzero_scalar();
-
-        // This can only fail due to internal errors, namely if k = 0 or s = 0 during signing which
-        // happens with negligible probability.
-        let sig = x
-            .try_sign_prehashed_rfc6979::<sha2::Sha256>(
-                *FieldBytes::<NistP256>::from_slice(H::digest(msg).as_ref()),
-                &[],
-            )
-            .unwrap()
-            .0;
+
+        let sig: Signature = self
+            .secret
+            .privkey
+            .sign_prehash(H::digest(msg).as_ref())
```

Comment:
is this the deterministic rfc6979 signature?

Reply:
Yes, it is. I'll add that to the docs to clarify.